spark/src/main/java/com/netflix/iceberg/spark/source/Writer.java

```diff
@@ -423,7 +356,7 @@ public void abort() throws IOException {
       if (currentAppender != null) {
         currentAppender.close();
         this.currentAppender = null;
-        fileIo.deleteFile(currentPath);
+        fileIo.deleteFile(currentFile.location());
```

Comment:
Maybe it would make sense to add `deleteFile(InputFile file)` with a default implementation that calls `file.location()`.